lib/commands/crafter/add_office.rb

```diff
@@ -21,41 +21,46 @@ def prepare(data)
       @apprentice = Apprentice.profile(data[:user_id])
     end
 
-    def parse_office_location(data)
-      data[:user_message].gsub("office:", "").strip.downcase
+    def run
+      return ""                 unless user
+      return office_unavailable unless @office.available?
+
+      if @crafter
+        @crafter.add_office(@office.location)
+        log_office_added
+      end
+
+      if @apprentice
+        @apprentice.add_office(@office.location)
+        log_office_added
+      end
+
+      "You were added to the office: #{@office.location}"
     end
 
-    def updated_crafter
-      "#{Crafter.with_office.count}/#{Crafter.all.count}"
+    private
+
+    def user
+      @crafter || @apprentice
     end
 
-    def updated_apprentice
-      "#{Apprentice.with_office.count}/#{Apprentice.all.count}"
+    def log_office_added
+      Logger.info(
+        "#{@office.location} was added to #{user.user_name} - #{updated_user}"
+      )
     end
 
-    def run
-      if @office.available? && (@crafter || @apprentice)
-        if @crafter
-          @crafter.add_office(@office.location)
-          Logger.info(
-            "#{@office.location} was added to #{@crafter.user_name} - #{updated_crafter}"
-          )
-        end
-
-        if @apprentice
-          @apprentice.add_office(@office.location)
-          Logger.info(
-            "#{@office.location} was added to #{@apprentice.user_name} - #{updated_apprentice}"
-          )
-        end
-
-        "You were added to the office: #{@office.location}"
-      elsif @crafter || @apprentice
-        Logger.info("#{@crafter.user_name} used an unavailable office: #{@office.location}")
-        "The offices available are: #{Office.locations.map(&:capitalize).join(", ")}"
-      else
-        ""
-      end
+    def office_unavailable
+      Logger.info("#{@crafter.user_name} used an unavailable office: #{@office.location}")
+      "The offices available are: #{Office.locations.map(&:capitalize).join(", ")}"
+    end
+
+    def updated_user
+      "#{user.class.with_office.count}/#{user.class.all.count}"
+    end
+
+    def parse_office_location(data)
```

Comment:
Yeah, I went through quite a few of those. At some point, I think that should be part of where we are reformating the data we receive from Slack, but not part of the commands

Reply:
yeah probably a good idea to have a `CommandSanitizer` or something that will clean up and format the incoming raw commands before they are passed on to the rest of the app.